OpenRA.Mods.Common/Warheads/HealthPercentageDamageWarhead.cs

```diff
@@ -42,6 +42,9 @@ public override void DoImpact(WPos pos, Actor firedBy, IEnumerable<int> damageMo
 
 		public override void DoImpact(Actor victim, Actor firedBy, IEnumerable<int> damageModifiers)
 		{
+			if (!IsValidAgainst(victim, firedBy))
+				return;
```

Comment:
It does not call `DamageWarhead`'s relevant `DoImpact` and goes directly into `InflictDamage`. See https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Warheads/SpreadDamageWarhead.cs#L76 vs https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Warheads/HealthPercentageDamageWarhead.cs#L51